docs/source/developers/release.rst

```diff
@@ -214,11 +211,16 @@ After the release vote, we must undertake many tasks to update source artifacts,
 
 Be sure to go through on the following checklist:
 
-#. Make the released version as "RELEASED" on JIRA
+#. Update the released milestone Date and set to "Closed" on GitHub
 #. Make the CPP PARQUET related version as "RELEASED" on JIRA
-#. Start the new version on JIRA on the ARROW project
 #. Start the new version on JIRA for the related CPP PARQUET version
 #. Merge changes on release branch to maintenance branch for patch releases
+
+.. code-block::
+    git checkout maint-<version>
+    git merge release-<version>-rc<rc-number>
+    git push apache maint-<version>
+
```

Comment:
We already have this in the following `dropdown`s:

```suggestion
```

Reply:
Thanks @kou I missed the dropdown below :)